File: app/src/main/java/com/ty/zbpet/ui/adapter/material/MaterialDiffUtil.java
package com.ty.zbpet.ui.adapter.material;

import android.support.annotation.Nullable;
import android.support.v7.util.DiffUtil;
import android.text.TextUtils;

import com.ty.zbpet.bean.material.MaterialDetailsIn;

import java.util.List;

/**
 * @author TY on 2018/11/27.
 * <p>
 * Different
 */
public class MaterialDiffUtil extends DiffUtil.Callback {

    private List<String> mOldList, mNewList;

    public MaterialDiffUtil(List<String> mOldList, List<String> mNewList) {
        this.mOldList = mOldList;
        this.mNewList = mNewList;
    }


    @Override
    public int getOldListSize() {
        return mOldList == null ? 0 : mOldList.size();
    }

    @Override
    public int getNewListSize() {
        return mNewList == null ? 0 : mNewList.size();
    }

    /**
     * 比较 ID
     *
     * @param i
     * @param i1
     * @return
     */
    @Override
    public boolean areItemsTheSame(int i, int i1) {
        return mOldList.get(i).getClass().equals(mNewList.get(i1).getClass());
    }

    /**
     * 比较 内容
     *
     * @param i
     * @param i1
     * @return
     */
    @Override
    public boolean areContentsTheSame(int i, int i1) {

        String oldStr = mOldList.get(i);
        String newStr = mNewList.get(i1);

        return oldStr.equals(newStr);
    }

    /**
     * 需要更新的数据
     *
     * @param oldItemPosition
     * @param newItemPosition
     * @return
     */
//    @Nullable
//    @Override
//    public Object getChangePayload(int oldItemPosition, int newItemPosition) {
//
//
//
//
//
//        return super.getChangePayload(oldItemPosition, newItemPosition);
//    }
}

File: app/src/main/java/com/ty/zbpet/ui/adapter/diffadapter/MaterialDiffUtil.kt
package com.ty.zbpet.ui.adapter.diffadapter

import android.support.v7.util.DiffUtil

/**
 * @author TY on 2018/11/27.
 *
 *
 * Different 扫码更新列表(箱码绑定)
 */
class MaterialDiffUtil(private val mOldList: List<String>?, private val mNewList: List<String>?) : DiffUtil.Callback() {


    override fun getOldListSize(): Int {
        return mOldList?.size ?: 0
    }

    override fun getNewListSize(): Int {
        return mNewList?.size ?: 0
    }

    /**
     * 比较 ID
     *
     * @param i
     * @param i1
     * @return
     */
    override fun areItemsTheSame(oldItemPosition: Int, newItemPosition: Int): Boolean {
        return mOldList!![oldItemPosition].javaClass == mNewList!![newItemPosition].javaClass
    }

    /**
     * 比较 内容
     *
     * @param i
     * @param i1
     * @return
     */
    override fun areContentsTheSame(oldItemPosition: Int, newItemPosition: Int): Boolean {

        val oldStr = mOldList!![oldItemPosition]
        val newStr = mNewList!![newItemPosition]

        return oldStr == newStr
    }

    /**
     * 需要更新的数据
     *
     * @param oldItemPosition
     * @param newItemPosition
     * @return
     */
    //    @Nullable
    //    @Override
    //    public Object getChangePayload(int oldItemPosition, int newItemPosition) {
    //
    //
    //
    //
    //
    //        return super.getChangePayload(oldItemPosition, newItemPosition);
    //    }
}
